### PR TITLE
Fix refillsRemaining to have unique count on each fill

### PR DESCRIPTION
**Pull request by kivarsson-va**
Co-authored-by: Joshua Nathaniel Miller <jmiller@rise8.us>
Co-authored-by: Kira Ivarsson <kivarsson@rise8.us>

### CODE REVIEW

**Comment by bschofield-va** [comment]
I need a little background information on this reason or motivation for this change. What problem is this solving? Can we get some test cases that show how to reproduce the undesired behavior?

**Comment by kivarsson-va** [comment]
@bschofield-va we pushed up an additional test to hopefully make the issue clearer. Each MedicationDispense should have a different number of refills remaining, i.e. if you have 3 refills left and you pick one up, you should now only have 2 refills left. The original logic calculated the same number of refills remaining for every dispense if the `fillsRemaining` field from VistA was not populated. The new logic ensures that the number of refills remaining is decremented on each fill, in order.

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/medicationdispense/R4SiteMedicationDispenseControllerTest.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import static gov.va.api.health.vistafhirquery.service.controller.MockRequests.json;
 import static gov.va.api.health.vistafhirquery.service.controller.MockRequests.requestFromUri;
 import static gov.va.api.health.vistafhirquery.service.controller.appointment.AppointmentSamples.R4.link;
+import static gov.va.api.health.vistafhirquery.service.controller.medicationdispense.MedicationDispenseSamples.*;
```
Comment (ilaflamme-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/medicationdispense/R4SiteMedicationDispenseControllerTest.java:6]: `import.*` is usually way more overhead than individual imports.

**Review by jhulbert-va — APPROVED** [review]
I made a little change to try and help with readability. Otherwise, I think it looks good based on my understanding of what it's trying to accomplish.

**Comment by bschofield-va** [comment]
Can I please have a request that shows this bug against the lab?

**Comment by kivarsson-va** [comment]
@bschofield-va we're not quite sure what you mean, can you elaborate?

**Comment by bschofield-va** [comment]
I'm not keen to take changes where I don't fully understand the bug and consequences of the change. Looking at this PR, I'm not convinced it works, but's very hard to evaluate solution without understanding the problem. I'd like to see bug in action, along with something that tells me what the right answer should be.

Perhaps you can fill in the blanks.`curl https://sandbox-api.va.gov/services/clinical-fhir/v0/r4/MedicationRequest?__________`

**Comment by treynolds-va** [comment]
Since the `fillsRemaining` field is on the rpcMed, it is a constant for every fill existing on that med. This change relies on the `fillsAllowed` field to index through each fill on the med. 

For example, if a med has 3 refills allowed, the first MedicationDispense will have 3 refills remaining, the second MedicationDispense will have 2 refills remaining, and so on. 

The previous implementation using the `fillsRemaining` field would transform with every MedicationDispense having the same value for refills remaining as it is the value of refills remaining for the entire med, not each individual fill.

**Comment by treynolds-va** [comment]
Bug in action using test patient:

Before changes:
Screen Shot 2022-08-08 at 2 03 21 PM

After changes:
Screen Shot 2022-08-08 at 1 54 53 PM

**Comment by bschofield-va** [comment]
This implementation is returning inconsistent results depending on query parameters. My team has evaluated the RPC, resulting XML and source Vista files and have a solution that should sort it out. I'm going to close this PR for now and re-open in the upcoming sprint after we've address the issues


```
$ curl -s http://localhost:8095/hcs/673/r4/MedicationDispense?patient=$P\&whenprepared=gt2000 | jq '.entry[].resource|select(.authorizingPrescription[0].reference == "http://localhost:8095/hcs/673/r4/MedicationRequest/sN5000000347-673-M33448")|[.id,.whenPrepared,.authorizingPrescription[0].reference,.extension]'
[
  "sN5000000347-673-M33448.3080728",
  "2008-07-28T00:00:00Z",
  "http://localhost:8095/hcs/673/r4/MedicationRequest/sN5000000347-673-M33448",
  [
    {
      "url": "http://hl7.org/fhir/StructureDefinition/medicationdispense-refillsRemaining",
      "valueInteger": 3
    }
  ]
]
[
  "sN5000000347-673-M33448.3110507",
  "2011-05-07T00:00:00Z",
  "http://localhost:8095/hcs/673/r4/MedicationRequest/sN5000000347-673-M33448",
  [
    {
      "url": "http://hl7.org/fhir/StructureDefinition/medicationdispense-refillsRemaining",
      "valueInteger": 2
    }
  ]
]
```
```
$ curl -s http://localhost:8095/hcs/673/r4/MedicationDispense?patient=$P\&whenprepared=gt2008 | jq '.entry[].resource|select(.authorizingPrescription[0].reference == "http://localhost:8095/hcs/673/r4/MedicationRequest/sN5000000347-673-M33448")|[.id,.whenPrepared,.authorizingPrescription[0].reference,.extension]'
[
  "sN5000000347-673-M33448.3110507",
  "2011-05-07T00:00:00Z",
  "http://localhost:8095/hcs/673/r4/MedicationRequest/sN5000000347-673-M33448",
  [
    {
      "url": "http://hl7.org/fhir/StructureDefinition/medicationdispense-refillsRemaining",
      "valueInteger": 3
    }
  ]
]
```

Note `sN5000000347-673-M33448.3110507` `refillsRemaining` values changes

**Pull request closed without merging**